### ISSUE TITLE
calculate_HMx power spectrum subroutine

### ISSUE THREAD

**Issue by alexander-mead** [alexander-mead/HMx#3]
Tilman, I've been messing around today and re-jigged the code a bit. It's not quite like we talked about this morning, but maybe close.

There is now a new subroutine 'calculate_HMx', which takes as its arguments 'itype' (2 integers that determine the 'type' of power to compute; if you look at the 'halo_type' names this tells you what the things you can compute are; -1 -1 would be a DMONLY power spectrum, 0 6 would be matter x pressure) a 'k' array, a 'a' (scale-factor) array and the cosmology. It's output is arrays of 'linear power', 'two halo power', 'one halo power' and 'non-linear power'.

An example of how this is used is when imode=2, so the 'k' and 'a' arrays are filled and then passed to 'calculate_HMx'. Is this the type of thing you can work with to integrate into CosmoSIS?

**Comment by alexander-mead** [comment]
Also, I added the 'parameters' to the cosmology type. This is a bit lazy, but it was the quickest thing to do. I will probably give them their own type eventually.

**Comment by tilmantroester** [comment]
I've done a couple changes (nothing really big but because I had to shuffle things around, alot of lines "changed") see #4 
Going forward it would be good to have only constants in the HMx module and pass the parameters explicitly. But maybe that's just my disdain for global variables.

**Comment by alexander-mead** [comment]
Okay, l

**Comment by alexander-mead** [comment]
I'm going to leave this issue 'open' for the time being. Not sure if this is a useful place for us to have conversations? Maybe.

**Comment by alexander-mead** [comment]
Okay, I merged your changes into my master branch. Thanks very much for those. Particularly things like the makefile I'd been meaning to generate for ages. Here are some comments:

In future, could you comment round any significant changes you make with '!TILMAN: Some comment' and '!TILMAN: End'. This will just help me keep track of things as time passes.

I'm sad you don't like the GOTO statement,  but I'll stick with your new GOTO-free version for the time being. I know there are lots of hardened computer programmers who think GOTO is like a tool of satan.

I've changed the names of the Makefiles around so we have 'Makefile -> Makefile_cosmosis' and 'Makefile_nocosmosis -> Makefile' just so that my one remains the default choice.

I've added some extra lines to the new 'Makefile' with my standard compilation flags. I also added a new 'debug' mode with some extra flags.

**Comment by tilmantroester** [comment]
You can see who did what with git, e.g., when you click on "Blame" on github. I feel that adding our name to changes would clutter up the code things quickly. I'll try to keep the commits more atomic (i.e. only one functional change per commit), so that it's easier to track what was changed when (and the commit message should say why).

**Comment by alexander-mead** [comment]
Hi Tilman,

I just did a bit of an overhaul of things and moved lots of the often-used subroutines and functions into their own module files (which I use for all my code). I will try to modularise the rest of the code in the coming few days. I also changed the Makefile so that it can compile the newly-modularised code. I'm afraid I may have butchered the nice Makefile that you set up, but let me know.
